benchmarks/benchmarks-core/src/jmh/java/io/micrometer/benchmark/core/MeterRegistrationBenchmark.java

```diff
@@ -17,44 +17,69 @@
 
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
-import org.openjdk.jmh.annotations.*;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
 import org.openjdk.jmh.runner.Runner;
 import org.openjdk.jmh.runner.RunnerException;
 import org.openjdk.jmh.runner.options.Options;
 import org.openjdk.jmh.runner.options.OptionsBuilder;
 
-import java.util.concurrent.TimeUnit;
-
-@State(Scope.Benchmark)
+@Measurement(iterations = 5, time = 1)
+@Warmup(iterations = 10, time = 1)
+@Fork(3)
+@BenchmarkMode(Mode.SampleTime)
 @OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Thread)
 public class MeterRegistrationBenchmark {
-    public static void main(String[] args) throws RunnerException {
-        Options opt = new OptionsBuilder()
-                .include(MeterRegistrationBenchmark.class.getSimpleName())
-                .warmupIterations(2)
-                .measurementIterations(5)
-                .mode(Mode.SampleTime)
-                .timeUnit(TimeUnit.SECONDS)
-                .forks(1)
-                .build();
+    MeterRegistry registry = new SimpleMeterRegistry();
 
-        new Runner(opt).run();
+    /** Ensures we are testing a clean registry each time */
+    @TearDown(Level.Invocation)
```

Comment:
`Level.Invocation` has a lot of warning flags in the javadoc. I'm not sure how warranted they are, but it's why I'm not sure if there's a great way of testing the case of no-redundancy without resulting in an unrealistically huge map.

Reply:
is the huge map really a problem? I'm wondering if the cure isn't worse than the disease here. If we just let it build up and clean per iteration vs per invocation like it was before.

Reply:
yet another option is a meter registry per test (like was before), but we'd have to sortof subtract out the overhead of that which in practice would not be initialized at counter scope.

Reply:
I don't think it's a huge problem for this benchmark - actually I was expecting you to just add comments to the previous version explaining it may not be representative of real-world but is how we're testing non-redundant registration here. The switch to `Level.Invocation` was a surprise ;)

Reply:
yeah I misunderstood you :P I don't mind dropping the last commit and doing that as it wasn't much work anyway

Reply:
While benchmarking the registration of one or two metrics in an empty MeterRegistry does tell us something, we probably also want to see how the registration is affected as the registry fills up. Also, from the big scary warnings in the JavaDoc:

> This level is only usable for benchmarks taking more than a millisecond

Hopefully we're not ever taking that long.
I'm also thinking maybe we go back to unbounded registration and hope that we don't OOM ourselves on an iteration.